### PR TITLE
Update to new circuits mantapay-v1

### LINKED ISSUE #812: Integrate MantaPay V1 changes into production/mock runtimes

### PR DESCRIPTION
**Pull request by Apokalip**
## Description
Changing structs and fixing encodes decodes relative to manta-rs changes and sdk needed changes

##Encode/Decode changes, some of the parity_scale_codec encodes decodes have been removed from manta-rs as custom trait implemenations are a headache needing to use scale_codec errors and not doing more than serde's serialization/deserialization. Most of the struct params are Fields/Groups that can be represented by a byte array [u8; 32] easily.

---

Before we can approve this PR for merge, please make sure that **all** the following items have been checked off:
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Added **one** label out of the `L-` group to this PR
- [x] Added **one or more** labels from the `A-` and `C-` groups to this PR
- [x] Explicitly labelled `A-calamari`, `A-dolphin` and/or `A-manta` if your changes are meant for/impact either of these (CI depends on it)
- [x] This PR is targeted against the *current*  Milestone ( otherwise discuss if it can be added in time)
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

Situational Notes:
- If adding functionality, write unit tests!
- If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.
- If runtime changes, need to update the version numbers properly:
   * `authoring_version`: The version of the authorship interface. An authoring node will not attempt to author blocks unless this is equal to its native runtime.
   * `spec_version`: The version of the runtime specification. A full node will not attempt to use its native runtime in substitute for the on-chain Wasm runtime unless all of spec_name, spec_version, and authoring_version are the same between Wasm and native.
   * `impl_version`: The version of the implementation of the specification. Nodes are free to ignore this; it serves only as an indication that the code is different; as long as the other two versions are the same then while the actual code may be different, it is nonetheless required to do the same thing. Non-consensus-breaking optimizations are about the only changes that could be made which would result in only the impl_version changing.
   * `transaction_version`: The version of the extrinsics interface. This number must be updated in the following circumstances: extrinsic parameters (number, order, or types) have been changed; extrinsics or pallets have been removed; or the pallet order in the construct_runtime! macro or extrinsic order in a pallet has been changed. You can run the `metadata_diff.yml` workflow for help. If this number is updated, then the `spec_version` must also be updated
- Verify benchmarks & weights have been updated for any modified runtime logics

### CODE REVIEW

File: pallets/manta-pay/src/lib.rs
Comment (Dengjianping) [inline pallets/manta-pay/src/lib.rs:68]: In this branch `feat/mantapay-v1`, is `utxo::v3` equivalent to `utxo::`?

File: pallets/manta-pay/src/lib.rs
Comment (Dengjianping) [inline pallets/manta-pay/src/lib.rs]: Not sure `expect` what it will affect if `expect` happens.
Or remove this `expect` with `map_err(...)?`

File: pallets/manta-pay/src/types.rs
Comment (Dengjianping) [inline pallets/manta-pay/src/types.rs]: Use `TryFrom` instead of `From` to handle the `expect`?

File: pallets/manta-pay/src/types.rs
Comment (Dengjianping) [inline pallets/manta-pay/src/types.rs]: `TryFrom`?

File: pallets/manta-pay/src/types.rs
Comment (Dengjianping) [inline pallets/manta-pay/src/types.rs]: `TryFrom`?

File: pallets/manta-pay/src/types.rs
Comment (Dengjianping) [inline pallets/manta-pay/src/types.rs]: `TryFrom`?

File: pallets/manta-pay/src/types.rs
Comment (Dengjianping) [inline pallets/manta-pay/src/types.rs]: `TryFrom`?

File: pallets/manta-pay/src/types.rs
Comment (Dengjianping) [inline pallets/manta-pay/src/types.rs]: `TryFrom`?

File: pallets/manta-pay/src/types.rs
Comment (Dengjianping) [inline pallets/manta-pay/src/types.rs]: `TryFrom`?

File: pallets/manta-pay/src/types.rs
Comment (Dengjianping) [inline pallets/manta-pay/src/types.rs]: `TryFrom`?

File: pallets/manta-pay/src/types.rs
Comment (Dengjianping) [inline pallets/manta-pay/src/types.rs]: `TryFrom`?

File: pallets/manta-pay/src/types.rs
Comment (Dengjianping) [inline pallets/manta-pay/src/types.rs]: `TryFrom`?

File: pallets/manta-pay/src/types.rs
Comment (Dengjianping) [reply pallets/manta-pay/src/types.rs]: If the rest of `From`s are reasonable, resolve my comments.

File: pallets/manta-pay/src/lib.rs
Comment (zqhxuyuan) [reply pallets/manta-pay/src/lib.rs:68]: there is no `utxo::v3` anymore in this branch, now just `utxo::`.

File: pallets/manta-pay/src/lib.rs
Comment (ghzlatarev) [inline pallets/manta-pay/src/lib.rs]: add comment why the new check

**Comment by ghzlatarev** [comment]
Tests are not compiling.

File: pallets/manta-pay/src/types.rs
Comment (ghzlatarev) [inline pallets/manta-pay/src/types.rs:76]: can you add an explanation why we are changing these in the description of the PR, please

Comment (Apokalip) [reply pallets/manta-pay/src/types.rs:76]: Do you mean the decodes/encodes?

Comment (ghzlatarev) [reply pallets/manta-pay/src/types.rs:76]: Yeah

File: pallets/manta-pay/src/lib.rs
Comment (ghzlatarev) [reply pallets/manta-pay/src/lib.rs]: +1 to error handle, otherwise add an explanation why this is impossible to hit,

File: pallets/manta-pay/src/lib.rs
Comment (ghzlatarev) [inline pallets/manta-pay/src/lib.rs]: can we error handle this

File: pallets/manta-pay/src/lib.rs
Comment (zqhxuyuan) [reply pallets/manta-pay/src/lib.rs]: done for all method except `register_all` and `spend_all` because the method return type is `()`.

File: pallets/manta-pay/src/lib.rs
Comment (ghzlatarev) [reply pallets/manta-pay/src/lib.rs]: what about the unwrap also ?

**Review by ghzlatarev — DISMISSED** [review]

**Review by ghzlatarev — DISMISSED** [review]

**Review by Dengjianping — DISMISSED** [review]

File: pallets/manta-pay/Cargo.toml
Comment (Dengjianping) [inline pallets/manta-pay/Cargo.toml]: Maybe it belongs to this PR, I'm just curious about why not use a tag or another branch, `main` branch would be not that reasonable. `main` receives merged PRs.

If there's any problem from manta-rs, we have to go to lock file to get the commit of manta-rs.

Comment (Garandor) [reply pallets/manta-pay/Cargo.toml]: There will be a new `manta-rs` release: https://github.com/Manta-Network/manta-rs/pull/290

Let's not block this until they're done, merge when ready and add a followup issue to staple to the version tag when it's done

File: pallets/manta-pay/src/types.rs
Comment (ghzlatarev) [inline pallets/manta-pay/src/types.rs]: shouldn't this be renamed to proof

File: node/src/builder.rs
Comment (ghzlatarev) [inline node/src/builder.rs]: There seems to be some rogues commits with irrelevant code like builder.rs

File: node/src/chain_specs/calamari.rs
Comment (ghzlatarev) [inline node/src/chain_specs/calamari.rs]: more rogue files.

File: node/src/chain_specs/dolphin.rs
Comment (ghzlatarev) [inline node/src/chain_specs/dolphin.rs]: more rogue files.

File: node/src/command.rs
Comment (ghzlatarev) [inline node/src/command.rs]: more

File: node/src/lib.rs
Comment (ghzlatarev) [inline node/src/lib.rs]: more

File: node/src/service.rs
Comment (ghzlatarev) [inline node/src/service.rs]: more

File: pallets/manta-pay/src/types.rs
Comment (ghzlatarev) [inline pallets/manta-pay/src/types.rs]: maybe name these types.

File: node/src/builder.rs
Comment (Apokalip) [reply node/src/builder.rs]: This seems like a mistake from merging manual seal mode into this so we can run it solo and someone forgot to revert it. Will clean that up.

File: pallets/manta-pay/src/lib.rs
Comment (ghzlatarev) [inline pallets/manta-pay/src/lib.rs]: This is decoding but emitting FP_ENCODE error

**Comment by zqhxuyuan** [comment]
done on https://github.com/Manta-Network/Manta/pull/911

**Pull request closed without merging**
